File: 3tip/3tip_php/cw4/models/Student.php
<?php
class Student extends Person
{

    public function __construct(
        protected int $id = -1,
        protected string $firstName = '',
        protected string $lastName = '',
        protected int $age = 0,
        protected string $studentId = '',
        protected float $gAvg = 10.0
    ) {
        // Wywołanie konstruktora klasy bazowej (Person) ustawia właściwości odziedziczone
        parent::__construct($id,$firstName, $lastName, $age);
        // Ustawienie właściwości specyficznych dla Student
        // $this->studentId = $studentId;
        // $this->gAvg = 67;
    }
    public function getStudentId(): string
    {
        return $this->studentId;
    }
    public function getGAvg(): float
    {
        return $this->gAvg;
    }
    // Nadpisanie metody __toString(), aby uwzględnić właściwości Studenta
    public function __toString(): string
    {
        return parent::__toString() . ", ID Studenta: " . $this->studentId
            . ", Średnia: " . $this->gAvg;
    }
}

File: 2tip/2tip_desk/cw11/Models/Student.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace cw11.Models
{
    public class Student
    {
        public int Id { get; set; }
        public string Firstname { get; set; }
        public string Lastname { get; set; }
        public int Age { get; set; }
        public int DivisionId { get; set; }
    }
}
